andela/haven-ah-backend
# #161342851 List All Non-admin Users

#### What does this PR do?
- Create List-all-users functionality

#### Description of Task to be completed?
- Refactor the signup route
- Check JWT authentication
- Create the list all user controller and route

#### How should this be manually tested?
- git clone `https://github.com/andela/haven-ah-backend.git`.
- Install `node` if not installed.
- ensure that PostgreSQL is running on your local machine.
- run `npm install` to install the npm packages.
- run `npm test` to run the unit and integration tests.
To test on postman
- run `npm run start-dev`
- On postman run GET `localhost:5000/api/users` (set JWT token as`x-access-token` in header).


#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#161342851

#### Screenshots (if appropriate)
Good Response:
<img width="993" alt="screen shot 2018-10-28 at 1 20 53 pm" src="https://user-images.githubusercontent.com/31534129/47615788-5a5fdb00-dab4-11e8-8e33-3925fc1f22df.png">

Bad Token Response:
<img width="983" alt="screen shot 2018-10-28 at 1 21 51 pm" src="https://user-images.githubusercontent.com/31534129/47615804-a1e66700-dab4-11e8-910b-8a85dbf33da7.png">

When token is not provided:
<img width="985" alt="screen shot 2018-10-28 at 1 22 20 pm" src="https://user-images.githubusercontent.com/31534129/47615807-bc204500-dab4-11e8-866b-4b5495683f81.png">
